### DISCUSSION TITLE
RLS with Supabase Views

### DISCUSSION THREAD

**Discussion by AlexanderMoonBit** [supabase/supabase#9110] [Questions]
Hi there,

We have quite a few views and just realized that views have admin rights and will bypass RLS.

What is the suggested solution to protect views to only return data related to that uid()? I'm worried someone will make a request to a view endpoint and get data not-related to them.

Would it be as simple as adding to the query uid() = user_id?

Cheers!

**Comment by steve-chavez** [comment]
Check this previous discussion: https://github.com/supabase/supabase/discussions/1501#discussioncomment-715670
